numpy/core/src/umath/ufunc_object.c

```diff
@@ -2865,35 +2865,44 @@ PyUFunc_Reduce(PyUFuncObject *ufunc, PyArrayObject *arr, PyArrayObject *out,
 
     /* Get the identity */
     identity = _get_identity(ufunc, &reorderable);
-    if (identity == NULL) {
+    if (initializer == NULL) {
+        initializer = identity;
+    }
+    else {
+        Py_DECREF(identity);
+        Py_INCREF(initializer);  /* match the reference count in the of above */
+    }
+
+
+    if (initializer == NULL) {
         return NULL;
     }
```

Comment:
This needs to happen _before_ you do the swap, otherwise `identity` might have been `NULL` and `DECREF`'d, which would segfault

Reply:
I've done it (and no tests fail locally), but just so I understand: Does `NULL` mean that something went wrong and the reduction failed?

Reply:
If a C function returns `NULL`, that usually means it set a python exception. If that happens, the right thing to do is usually to clean up, and then return `NULL` all the way back to the calling function.

Have a look at `_get_identity` to see how it can return `NULL`.